src/main/java/com/aws/greengrass/clientdevices/auth/configuration/CAConfiguration.java

```diff
@@ -114,6 +114,16 @@ private static CertificateStore.CAType getCaTypeFromConfiguration(Topics configu
         return CertificateStore.CAType.valueOf(caType);
     }
 
+    private static URI getUriOrThrow(String maybeUri) throws URISyntaxException {
+        URI maybeValid = new URI(maybeUri);
+
+        if (maybeValid.getScheme() == null) {
+            throw new URISyntaxException(maybeUri, "Uri is missing the scheme");
+        }
+
+        return maybeValid;
```

Comment:
nit: similar nit to below, naming is confusing cause we're returning a valid uri.  I think using something along the lines of `Uri uri` and `String rawUri` would be fine.